pandas/tseries/tdi.py

```diff
@@ -344,6 +344,17 @@ def _add_datelike(self, other):
             other = Timestamp(other)
             i8 = self.asi8
             result = i8 + other.value
+            # gh-14068: there is the possibility of addition overflow,
+            # which occurs when we add two very large positive numbers,
+            # resulting in a negative number.
+            #
+            # Since Timestamp objects can never have a negative value,
+            # we just need to check that the elements in result are also
+            # nonnegative when their corresponding elements in i8 are.
+            for x, y in zip(i8, result):
```

Comment:
To that point overflow isn't guaranteed produce a negative number here - it is undefined behavior in C, so impl defined.  But it is possible to check for it, simpler version [here](http://c-faq.com/misc/intovf.html)

re: option -  it's never a good thing, just meant that you may not always want to pay the perf penalty for it, along the lines of bounds checking, and (presumably) why numpy doesn't do it on regular ints.